app/Http/Controllers/SessionController.php

```diff
@@ -56,75 +60,92 @@ public function __construct(ConfigFunctions $configFunctions, GitHubFunctions $g
 	 *
 	 * @throws ModelNotFoundException
 	 * @throws VersionControlException
+	 * @throws ConfigurationKeyMissingException
+	 * @throws InvalidOrderDirectionException
+	 * @throws FrameworkException
 	 */
 	public function init(): array
 	{
-		$logged_in = AccessControl::is_logged_in();
-
-		// Return settings
-		$return = [];
-
-		// Check if login credentials exist and login if they don't
-		if (AccessControl::noLogin() === true || $logged_in === true) {
-			// we set the user ID (it is set to 0 if there is no login/password = admin)
-			$user_id = AccessControl::id();
-
-			if ($user_id === 0) {
-				$return['status'] = Config::get('defines.status.LYCHEE_STATUS_LOGGEDIN');
-				$return['admin'] = true;
-				$return['may_upload'] = true; // not necessary
+		try {
+			$isLoggedIn = AccessControl::is_logged_in();
+			$isAdminNotRegistered = AccessControl::noLogin();
+
+			// Return settings
+			$return = [];
+
+			if ($isAdminNotRegistered) {
+				// TODO: Remove this legacy stuff after creating the admin user has become part of the installation routine.
+				// If the session is unauthenticated ('user' === null), but grants admin rights nonetheless,
+				// the front-end shows the dialog to create an admin account.
+				$return['user'] = null;
+				$return['rights'] = [
+					'is_admin' => true,
+					'is_locked' => false,
+					'may_upload' => true,
+				];
+			} elseif ($isLoggedIn) {
+				$user_id = AccessControl::id();
+				/** @var User $user */
+				$user = User::query()->findOrFail($user_id);
+				$return['user'] = $user->toArray();
+				$return['rights'] = [
+					'is_admin' => $user->is_admin(),
+					'is_locked' => $user->is_locked,
+					'may_upload' => $user->may_upload,
+				];
+			} else {
+				$return['user'] = null;
+				$return['rights'] = [
+					'is_admin' => false,
+					'is_locked' => true,
```

Comment:
why ?

Reply:
Why what?

Reply:
why `is_locked = true` ? 
What I mean is that it shouldn't even matter. :)

Reply:
We cannot leave it undefined, so we must either decide for `true` or `false`. Obviously, the anonymous user cannot change a password nor a username. So the safe default is `true`.